### PR TITLE
fix: filter dismissed repairs in overview and system_health (#1307)

### LINKED ISSUE #1307: ha_get_overview repairs field doesn't expose ignored / is_fixed state
## Problem

`ha_get_overview` returns a `repairs[]` array and `repair_count`, but the entries only include `issue_id`, `domain`, `severity`, and `translation_key`. The underlying HA `repairs/list_issues` WebSocket API also returns `ignored`, `dismissed_version`, and `is_fixed`, but those fields are dropped.

As a result, repairs the user has already dismissed in the HA UI still show up in the overview indistinguishably from active ones. An LLM consumer reading the overview will treat them as actionable and send the user chasing a problem they already resolved.

## Repro

1. Have a repair in HA that you've ignored via Settings → System → Repairs.
2. Call `ha_get_overview`.
3. The ignored repair appears in `repairs[]` and is counted in `repair_count`, with no flag indicating it's dismissed.

Example payload:
\`\`\`json
{
  "repair_count": 1,
  "repairs": [
    {
      "issue_id": "move_integration_to_ha_coremass",
      "domain": "mass",
      "severity": "warning",
      "translation_key": "move_integration_to_ha_core"
    }
  ]
}
\`\`\`
(This \`mass\` repair was ignored in HA 2025.12.3 but still surfaces.)

## Suggested fix

Either (or both):
- **Default-filter** ignored and \`is_fixed\` repairs out of \`repairs[]\` and \`repair_count\`, with an opt-in flag like \`include_dismissed_repairs=True\` to get them back.
- **Pass through** \`ignored\`, \`dismissed_version\`, and \`is_fixed\` fields on each repair entry so callers can filter themselves.

The filter-by-default option matches what a human sees in the Repairs UI and is probably the more useful default for LLM consumers.

### PR DESCRIPTION
**Pull request by kingpanther13**
## What does this PR do?

Closes #1307.

`ha_get_overview` and `ha_get_system_health(include="repairs")` both
returned every repair from HA's `repairs/list_issues` WebSocket API,
including ones the user had already dismissed via Settings → System →
Repairs. `ha_get_overview` additionally stripped the dismissal-state
fields (`ignored`, `dismissed_version`, `is_fixable`), so agents had no
way to distinguish active from dismissed repairs and would send users
chasing problems they had already resolved.

Both tools now:
- **Default-filter `ignored` repairs** to match the HA Repairs UI.
- Accept **`include_dismissed_repairs=True`** to opt back in.
- Report a separate `dismissed_repair_count` (overview) /
  `dismissed_count` (system_health) when there are hidden repairs, so
  agents can still surface that state if relevant.
- Pass through `ignored`, `dismissed_version`, `is_fixable`,
  `breaks_in_ha_version`, `created`, `issue_domain` on each repair so
  opt-in callers can introspect state.

Filtering and projection are centralized in `util_helpers`
(`filter_active_repairs`, `project_repair_fields`) so both call sites
stay in sync.

## Type of change
- [x] 🐛 Bug fix

## Testing
- [ ] I have tested these changes with a LLM agent
- [ ] All automated tests pass (`uv run pytest`)
- [x] Code follows style guidelines (`uv run ruff check`)

New unit tests cover both tools and the helper:
- `tests/src/unit/test_overview_repairs.py` — overview filtering, projection, opt-in, all-dismissed edge case
- `tests/src/unit/test_tools_system.py::TestFetchRepairs` — `_fetch_repairs` filtering, pass-through, dismissed counter
- `tests/src/unit/test_util_helpers.py` — `filter_active_repairs` + `project_repair_fields`

## Checklist
- [x] I have updated documentation if needed

### CODE REVIEW

File: src/ha_mcp/tools/tools_search.py [hunk 1]
```diff
@@ -873,6 +885,13 @@ async def ha_get_overview(
         include_notifications_bool = coerce_bool_param(
             include_notifications, "include_notifications", default=True
         )
+        include_dismissed_repairs_bool = bool(
+            coerce_bool_param(
+                include_dismissed_repairs,
+                "include_dismissed_repairs",
+                default=False,
+            )
+        )
```
[the remark this answers is not in the document]
Comment (kingpanther13) [reply src/ha_mcp/tools/tools_search.py:894]: 📝 Not addressing.

This file's pattern for `ha_get_overview` is to coerce all params (`coerce_int_param`, `parse_string_list_param`, sibling `coerce_bool_param` calls in lines 866–892) BEFORE the try block. Wrapping only this one in try/except diverges from local convention — and per your own guidance, "Prioritize consistency with established file-wide patterns over local hardening."

The `bool()` wrapper is pyright-driven, not redundant: `coerce_bool_param` is annotated `-> bool | None` (returns `None` when `default` is `None`), so pyright flagged the unwrapped value where downstream code expects `bool`. The proper fix is overloads on `coerce_bool_param` — separate refactor.

File: src/ha_mcp/tools/tools_system.py [hunk 2]
```diff
@@ -348,13 +349,21 @@ async def ha_get_system_health(
...
+        - include_dismissed_repairs: Include user-dismissed/ignored repairs (default: False). Only meaningful when "repairs" is in `include`.
         """
         includes = self._parse_includes(include)
+        include_dismissed_repairs_bool = bool(
+            coerce_bool_param(
+                include_dismissed_repairs,
+                "include_dismissed_repairs",
+                default=False,
+            )
+        )
```
[the remark this answers is not in the document]
Comment (kingpanther13) [reply src/ha_mcp/tools/tools_system.py:366]: 📝 Same response as the `tools_search.py:894` thread — declining for consistency with file-wide pattern; `bool()` wrapper is type-driven.

File: src/ha_mcp/tools/tools_system.py [hunk 3]
```diff
@@ -21,7 +21,7 @@
     raise_tool_error,
     register_tool_methods,
 )
-from .util_helpers import coerce_bool_param
+from .util_helpers import coerce_bool_param, filter_active_repairs
```
[the remark this answers is not in the document]
Comment (kingpanther13) [reply src/ha_mcp/tools/tools_system.py:24]: 📝 Not addressing — see the related thread on line 489 for the design rationale.

File: src/ha_mcp/tools/tools_system.py [hunk 4]
```diff
@@ -454,17 +466,31 @@ async def _fetch_health_info(self) -> tuple[Any, dict[str, Any]]:
...
+                all_issues = repairs_result.get("result", {}).get("issues", [])
+                visible_issues = filter_active_repairs(
+                    all_issues, include_dismissed=include_dismissed
+                )
                 repairs = {
-                    "issues": repairs_list,
-                    "count": len(repairs_list),
+                    "issues": visible_issues,
+                    "count": len(visible_issues),
                 }
```
[the remark this answers is not in the document]
Comment (kingpanther13) [reply src/ha_mcp/tools/tools_system.py:489]: 📝 Not addressing — this is a deliberate two-tier design:

- `ha_get_overview` is the summary tier. Projection (drops `translation_placeholders`, `learn_more_url`) keeps the overview payload compact, since repairs are part of a much larger system snapshot.
- `ha_get_system_health(include="repairs")` is the detail tier. Callers opt in specifically because they need the full payload — including `translation_placeholders`, which carries the human-readable context (which entities, which automation, edit URL) that lets an agent reason about *what* the repair actually is.

If we project here too, `ha_get_system_health` loses the only field that explains what the repair means, with no remaining tool that exposes it. The asymmetry is the point.

File: src/ha_mcp/tools/tools_search.py [hunk 5]
```diff
@@ -951,24 +970,29 @@ async def ha_get_overview(
...
+        # Include active repair issues. By default we filter out user-dismissed
+        # ("ignored") repairs to match the HA Repairs UI — surfacing them as
+        # actionable would send agents chasing problems the user already
+        # resolved. Set `include_dismissed_repairs=True` to get all repairs.
         result["repair_count"] = 0
         try:
             repairs_result = await client.send_websocket_message(
                 {"type": "repairs/list_issues"}
             )
             if repairs_result.get("success"):
```
[the remark this answers is not in the document]
Comment (kingpanther13) [reply src/ha_mcp/tools/tools_search.py:980]: ✅ Fixed in f050c68a — both `ha_get_overview` and `_fetch_repairs` now extract `error.message` from the `success: False` envelope and surface it via `repairs_error` / `error`, matching the existing exception-path behavior. Unit tests added for both branches.

**Review by kingpanther13 — COMMENTED** [review]
Addressed Gemini review feedback in f050c68a.

**Accepted (line 980):** `success: False` from `repairs/list_issues` now surfaces the API error message via `repairs_error` / `error` on both `ha_get_overview` and `ha_get_system_health(include="repairs")`. Plus new test coverage for the success-false branch, the WS-exception branch, and string-coerced boolean inputs to `include_dismissed_repairs`.

**Declined:**
- *try/except around param coercion* (lines 894, 366): all sibling param coercion in `ha_get_overview` happens outside the try block — wrapping just this one diverges from the file-wide pattern. The `bool()` wrapper is pyright-driven (`coerce_bool_param` is typed `-> bool | None`), not redundant.
- *Project fields in `_fetch_repairs`* (lines 24, 489): deliberate two-tier design — `ha_get_system_health(include="repairs")` is the detail tier and intentionally keeps `translation_placeholders` so agents can see *which* entities/automations are involved.

**Comment by kingpanther13** [comment]
## Implementation Summary

**Choices Made:**
- **Filter-by-default + opt-in flag**, not just pass-through. Reporter offered both options; default-filter matches the HA Repairs UI and is the right default for LLM consumers who interpret `repair_count` as "things to act on." `include_dismissed_repairs=True` keeps the full set available.
- **Two-tier repair payload**: `ha_get_overview` projects to a compact field subset (drops `translation_placeholders`, `learn_more_url`); `ha_get_system_health(include="repairs")` returns the full HA payload. Overview is the summary tier; system_health is the detail tier where callers opt in specifically to get the entity/automation context that `translation_placeholders` carries.
- **Surface `dismissed_repair_count` / `dismissed_count` only when non-zero**, so agents can spot hidden repairs without bloat on the common path.
- **Centralized helpers** (`filter_active_repairs`, `project_repair_fields`) in `util_helpers.py` so the two call sites can't drift.
- **Pass through `is_fixable`, not `is_fixed`**: HA's `repairs/list_issues` exposes `is_fixable` (whether the repair has an automated fix); there is no `is_fixed` field (resolved issues drop off the list entirely). Surfaced what HA actually returns.
- **Field projection includes `breaks_in_ha_version`, `created`, `issue_domain`** alongside the dismissal-state fields — small enough to keep, useful for agent triage of severity/recency.

**Problems Encountered:**
- **`coerce_bool_param` return type is `bool | None`** — pyright flagged the unwrapped value where downstream code expects `bool`. Worked around with a `bool()` wrapper; a cleaner fix is to add overloads on `coerce_bool_param` so a non-None default narrows the return type. Gemini flagged the wrapper as "redundant" — declined, since removing it re-introduces the pyright diagnostic.
- **Silent `success: False` handling** — the original code path checked `success` but did nothing on the failure branch, leaving `repair_count: 0` indistinguishable from "no repairs." Gemini caught this; fixed in f050c68a to surface `error.message` via `repairs_error` / `error`, matching the existing exception-path behavior.

**Review by Patch76 — APPROVED** [review]
The default-filter + opt-in shape matches #1307's ask, and extending the same behavior to `ha_get_system_health(include="repairs")` covers the symmetric leak on that path.

Test coverage is solid: 442 LOC of tests for 130 LOC of source — the dedicated `test_overview_repairs.py` exercises filter / opt-in / projection / all-dismissed / empty paths, and the helper-level coverage in `test_util_helpers.py` locks the projection contract independently of the call sites.

The two-tier projection design (`ha_get_overview` projects via `project_repair_fields`, `_fetch_repairs` returns raw payload for the detailed-tier consumer) is non-obvious from the source alone — your reply on the Gemini thread spells out the reasoning. A one-liner near `_fetch_repairs` would surface that design intent without re-reading the PR.
